modules/apps/commerce/headless/headless-commerce/headless-commerce-delivery-cart-impl/src/main/java/com/liferay/headless/commerce/delivery/cart/internal/resource/v1_0/CartResourceImpl.java

```diff
@@ -478,6 +499,144 @@ private CommerceOrder _upsertShippingAddress(
 			commerceOrder.getAdvanceStatus(), commerceContext);
 	}
 
+	private void _validateBillingAddressSelected(
+			CommerceOrder commerceOrder, List<Error> errors)
+		throws PortalException {
+
+		CommerceAddress billingAddress = commerceOrder.getBillingAddress();
+		CommerceAddress shippingAddress = commerceOrder.getShippingAddress();
+
+		if (billingAddress == null) {
+			Error error = new Error();
+
+			error.setErrorDescription("No billing address selected");
+
+			errors.add(error);
```

Comment:
@andrea-ale-sbarra we can probably extract this out to a private method since its being used in quite a few places in here. Maybe something like this.
```
private Error _addError(String description) {
    Error error = new Error();

    error.setErrorDescription(description);

    errors.add(error);

    return error;
}
```
so then we can do something like

```suggestion
			_addError("No billing address selected");
```